.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: Unit Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+
+# Cancel previous jobs if an update has been made to the pull request
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+env:
+  CLANG_DOCKER_IMAGE: seracllnl/tpls:clang-14_08-15-24_21h-49m
+  CUDA_DOCKER_IMAGE: seracllnl/tpls:cuda-12_08-15-24_21h-49m
+  GCC_DOCKER_IMAGE: seracllnl/tpls:gcc-13_08-15-24_21h-49m
+
+jobs:
+  # Hacky solution to reference env variables outside of `run` steps https://stackoverflow.com/a/74217028
```

Comment:
Good job figuring this out but man is this wonky... The fact they had to write this page to explain contexts and didn't fix it... sigh.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability

Reply:
there is a better way using variable contexts but it requires admin access to set those variables. and they would be somewhere in github settings, rather than in the yml.

Reply:
This is fine. If we put them in the github settings then the updates cannot be reviewed in PRs